nixos/modules/hardware/tuxedo-keyboard.nix

```diff
@@ -0,0 +1,42 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let 
+  cfg = config.tuxedo-keyboard;
+  tuxedo-keyboard = config.boot.kernelPackages.tuxedo-keyboard;
+in
+  {
+    options.tuxedo-keyboard = {
+      enable = mkOption {
+        type = types.bool;
+        default = false;
+
+        description = ''
+          Enables the tuxedo-keyboard driver.
+          
+          To configure the driver, pass the options to the <literal>boot.kernelParams</literal> configuration.
+          There are several parameters you can change. It's best to check at the source code description which options are supported.
+          You can find all the supported parameters at: <link xlink:href="https://github.com/tuxedocomputers/tuxedo-keyboard#kernelparam" />
+
+          In order to use the 'custom' lighting with the maximumg brightness and a color of '0xff0a0a' one would put pass <literal>boot.kernelParams</literal> like this:
+
+          <option>
+          boot.kernelParams = [
+           "tuxedo_keyboard.mode=0"
+           "tuxedo_keyboard.brightness=255"
+           "tuxedo_keyboard.color_left=0xff0a0a"
+          ];
+          </option>
```

Comment:
There's `<programlisting></programlisting>` to render multiple lines of code

Reply:
That's a good tip! :)